teamspeak3.kvp

```diff
@@ -0,0 +1,74 @@
+App.LinuxCommandLineArgs=
+App.WindowsCommandLineArgs= 
+App.CommandLineArgs={{$PlatformArgs}} inifile=teamspeak3server.ini license_accepted=1 {{$FormattedArgs}}
+App.AppSettings={}
+App.EnvironmentVariables={}
+App.CommandLineParameterFormat={0}={1}
+App.CommandLineParameterDelimiter= 
+App.ExitMethod=OS_CLOSE
+App.ExitTimeout=30
+App.ExitString=stop
+App.ExitFile=app_exit.lck
+App.HasWriteableConsole=True
+App.HasReadableConsole=True
+App.SupportsLiveSettingsChanges=False
+App.LiveSettingChangeCommandFormat=set {0} "{1}"
+App.ApplicationIPBinding=0.0.0.0
+App.Ports=[{"Protocol":"UDP","Port":9987,"Offset":0,"Range":1,"Ref":"Voice","Name":"Voice Port","Description":"Port for voice communications","ChildPorts":null},{"Protocol":"TCP","Port":30033,"Offset":0,"Range":1,"Ref":"Filetransfer","Name":"File Transfer Port","Description":"Port for file transfers","ChildPorts":null},{"Protocol":"TCP","Port":10011,"Offset":0,"Range":1,"Ref":"ServerQueryRAW","Name":"Telnet Port","Description":"Port for Telnet administration","ChildPorts":null},{"Protocol":"TCP","Port":10022,"Offset":0,"Range":1,"Ref":"ServerQuerySSH","Name":"SSH Port","Description":"Port for SSH administration","ChildPorts":null},{"Protocol":"TCP","Port":10080,"Offset":0,"Range":1,"Ref":"WebQueryHTTP","Name":"HTTP Port","Description":"Port for HTTP administration","ChildPorts":null},{"Protocol":"TCP","Port":10443,"Offset":0,"Range":1,"Ref":"WebQueryHTTPS","Name":"HTTPS Port","Description":"Port for HTTPS administration","ChildPorts":null}]
+App.AdminPortRef=ServerQueryRAW
+App.PrimaryApplicationPortRef=Voice
+App.UniversalSleepApplicationUDPPortRef=
+App.UniversalSleepSteamQueryPortRef=
+App.MaxUsers=32
+App.UseRandomAdminPassword=True
+App.RemoteAdminPassword=
+App.AdminMethod=TelnetRCON
+App.RCONConnectDelaySeconds=5
+App.RCONConnectRetrySeconds=5
+App.UpdateSources=[{"UpdateStageName":"TeamSpeak 3 Windows","UpdateSourcePlatform":"Windows","UpdateSource":"FetchURLFromJQ","UpdateSourceData":"https://teamspeak.com/versions/server.json","UpdateSourceArgs":"$['windows']['x86_64']['mirrors']['teamspeak.com']","UnzipUpdateSource":true,"OverwriteExistingFiles":true},{"UpdateStageName":"TeamSpeak 3 Linux","UpdateSourcePlatform":"Linux","UpdateSource":"FetchURLFromJQ","UpdateSourceData":"https://teamspeak.com/versions/server.json","UpdateSourceArgs":"$['linux']['x86_64']['mirrors']['teamspeak.com']","UnzipUpdateSource":true,"OverwriteExistingFiles":true},{"UpdateStageName":"Create Symlink","UpdateSourcePlatform":"Linux","UpdateSource":"Executable","UpdateSourceData":"/bin/ln","UpdateSourceArgs":"-s {{$FullBaseDir}}teamspeak3-server_linux_amd64 {{$FullBaseDir}}teamspeak3-server_win64"},{"UpdateStageName":"Config File Download","UpdateSourcePlatform":"All","UpdateSource":"FetchURL","UpdateSourceData":"https://github.com/IceOfWraith/AMPTemplates/raw/ts3/teamspeak3server.ini","UpdateSourceTarget":"{{$FullBaseDir}}/teamspeak3-server_win64/","OverwriteExistingFiles":false}]
```

Comment:
- Add DeleteAfterExtract to the download stage, to remove the archive after extract
- There is a rogue / after one of the {{$FullBaseDir}} refs
- Instead of the symlink, could you not rename the archive on download to a neutral name (eg teamspeak3-server) on both platforms, so that the dir after extract is the same?

Reply:
First two updated.

I used a symlink because inside the zip is a different folder structure, not the name of the zip itself. It's very short-sighted of the creators for automation. A rename of the folder would work too, but I don't recall that being a built in function. I intend to convert this to the built-in symlink function when Mike has that working.